src/BBT.StructureTools.Tests/Convert/ConverterIntTests.cs

```diff
@@ -0,0 +1,579 @@
+
+            var converter = GetConverter<SourceRoot, TargetRoot>();
+
+            var sourceRoot = new SourceRoot()
+            {
+                Tree = new SourceTree(),
+            };
+            var targetRoot = new TargetRoot();
+
+            // Act
+            var processings = new List<IBaseAdditionalProcessing>();
+            converter.Convert(sourceRoot, targetRoot, processings);
+
+            // Assert
+            targetRoot.TargetTree.Should().NotBeNull();
+            targetRoot.TargetTree.RelationOnTarget.Should().Be(targetRoot.RelationOnTarget);
+        }
+
+        /// <summary>
+        /// Tests the convert from root to leaf into target structure.
+        /// </summary>
+        [Fact]
+        public void Convert_RegisterCreateToManyWithRelation_Successful()
+        {
+            // Arrange
+            container.Bind<IConvertRegistrations<SourceTree, TargetTree, IForTest>>().To<CreateToManyWithRelationRegistrations>();
+            container.Bind<IConvertRegistrations<SourceTreeLeaf, TargetTreeLeaf, IForTest>>().To<CopyLeafAttributeRegistrations>();
+
+            var converter = GetConverter<SourceTree, TargetTree>();
+
+            var sourceTree = new SourceTree()
+            {
+            };
+
+            sourceTree.Leafs.Add(new SourceTreeLeaf());
+
+            var targetTree = new TargetTree()
+            {
+                RelationOnTarget = new MasterData(),
+            };
+
+            // Act
+            var processings = new List<IBaseAdditionalProcessing>();
+            converter.Convert(sourceTree, targetTree, processings);
+
+            // Assert
+            targetTree.TargetLeafs.Count.Should().Be(1);
+            var targetLeaf = targetTree.TargetLeafs.Single();
+            targetLeaf.RelationOnTarget.Should().Be(targetTree.RelationOnTarget);
+        }
+
+        #endregion
+
+        #region Asserts
+
+        private static void AssertRoot(SourceRoot sourceRoot, TargetRoot targetRoot)
+        {
+            targetRoot.OriginRoot.Should().Be(sourceRoot);
+            targetRoot.Name.Should().Be(sourceRoot.Name);
+            targetRoot.RootId.Should().Be(sourceRoot.Id);
+            targetRoot.Id.Should().Be(sourceRoot.Id);
+            targetRoot.TargetId.Should().Be(sourceRoot.Id);
+            targetRoot.NumberSourceDefault.Should().NotBe(sourceRoot.NumberSourceDefault);
+            targetRoot.NumberSourceNotDefault.Should().Be(sourceRoot.NumberSourceNotDefault);
+            targetRoot.NumberTargetDefault.Should().Be(sourceRoot.NumberTargetDefault);
+            targetRoot.NumberTargetNotDefault.Should().NotBe(sourceRoot.NumberTargetNotDefault);
+            targetRoot.NumberSourceLookedUp.Should().Be(sourceRoot.NumberSourceNotDefault);
+            targetRoot.NumberSourceNotLookedUp.Should().Be(sourceRoot.NumberSourceNotDefault);
+
+            // ToDo LIS: Check is this a bug?
```

Comment:
Please create an Issue for this and reference the issue number here.

Reply:
issue created and referenced.